### PR TITLE
[ISSUE-1057] bugfix upgrade go 1.21.

### LINKED ISSUE #1057: Remove CVEs from csi-baremetal images
**Describe the bug**
Removing all critical/high CVEs from images

**Environment (please complete the following information):**
 - Hardware: all
 - OS: all
 - Version: 1.4

**To Reproduce**
Run security scans

**Expected behavior**
No critical/high CVEs

### PR DESCRIPTION
**Pull request by Andrzej-Zukowski**
### Resolves #1057 (https://github.com/dell/csi-baremetal/issues/1057)

Library upgrades and base images

## PR checklist
- [ ] Add link to the issue
- [ ] Choose Project
- [ ] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
Full regression tests

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ on:
 
 env:
   REGISTRY: 'ghcr.io/dell/csi-baremetal'
-  go_version: '1.19.2'
+  go_version: '1.21.3'
   golangci_version: '1.48.0'
```
Comment (libzhang) [inline .github/workflows/e2e.yml:12]: I see other golangci-lint version changed to 1.55.2,  should we also change this together?

**Comment by libzhang** [comment]
please also change the PR title to [ISSUE-1057] bugfix upgrade go 1.21.

File: devkit/Makefile [hunk 2]
```diff
@@ -26,13 +26,13 @@ TAG  ?= $(major).$(minor).$(patch)-$(git_rev)
 IMAGE_REPO_TAG := $(REPO):$(TAG)
 
 DOCKER_VERSION ?= 20.10.14_ce
-GO_VERSION ?= 1.19.10
-GOLANGCI_VERSION ?= 1.48.0
+GO_VERSION ?= 1.21.3
+GOLANGCI_VERSION ?= 1.55.2
```
Comment (libzhang) [inline devkit/Makefile:30]: devkit change need to pass the multiple jenkins job,  (csi-build,  csi-kind-acceptance,  csi-ci), as these tests are based on the devkit image.

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (Andrzej-Zukowski) [reply .github/workflows/e2e.yml]: Done

File: devkit/Makefile (hunk 2 above)
Comment (Andrzej-Zukowski) [reply devkit/Makefile:30]: csi-build - PASSEDcsi-kind-acceptance, csi-ci

**Review by libzhang — APPROVED** [review]

**Review by katarzyna-kulpa — APPROVED** [review]

**Pull request merged**